error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/AssertJMapRules.java

```diff
@@ -171,13 +176,31 @@ AbstractBooleanAssert<?> before(Map<K, V> map, K key) {
       return assertThat(map.containsKey(key)).isFalse();
     }
 
+    @BeforeTemplate
+    AbstractCollectionAssert<?, Collection<? extends K>, K, ?> before2(Map<K, V> map, K key) {
+      return assertThat(map.keySet()).doesNotContain(key);
+    }
+
     @AfterTemplate
     @UseImportPolicy(STATIC_IMPORT_ALWAYS)
     MapAssert<K, V> after(Map<K, V> map, K key) {
       return assertThat(map).doesNotContainKey(key);
     }
   }
 
+  static final class AssertThatMapContainsOnlyKey<K, V> {
+    @BeforeTemplate
+    AbstractCollectionAssert<?, Collection<? extends K>, K, ?> before2(Map<K, V> map, K key) {
+      return assertThat(map.keySet()).containsOnly(key);
```

Comment:
Interestingly this has overlap with the rule we have for sets, but the AfterTemplate here is more specific for Maps. 🤔 
https://github.com/PicnicSupermarket/error-prone-support/blob/bce570f2d0626bade3b09ca764889512c543a3f8/error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/AssertJRules.java#L240-L251

Reply:
Indeed: our self-check flags that in this case the new check should match against `containsExactly`, as existing code is expected to have already been rewritten by the `AssertThatSetContainsExactlyOneElement` rule.